qiita_db/support_files/patches/python_patches/40.py

```diff
@@ -0,0 +1,22 @@
+from random import SystemRandom
+from string import ascii_letters, digits
+
+from qiita_db.sql_connection import TRN
+
+pool = ascii_letters + digits
+with TRN:
+    # 2 and 3 are the ids of the 2 new software rows, the BIOM and
+    # target gene type plugins
+    for i in [2, 3]:
+        client_id = ''.join([SystemRandom().choice(pool) for _ in range(50)])
```

Comment:
Awesome, thanks for the description!

Yoshiki Vázquez-Baeza

> On Mar 16, 2016, at 2:08 PM, Jose Navas <notifications@github.com> wrote:
> 
> In qiita_db/support_files/patches/python_patches/40.py:
> 
> > @@ -0,0 +1,22 @@
> > +from random import SystemRandom
> > +from string import ascii_letters, digits
> > +
> > +from qiita_db.sql_connection import TRN
> > +
> > +pool = ascii_letters + digits
> > +with TRN:
> > +    # 2 and 3 are the ids of the 2 new software rows, the BIOM and
> > +    # target gene type plugins
> > +    for i in [2, 3]:
> > +        client_id = ''.join([SystemRandom().choice(pool) for _ in range(50)])
> In theory, these steps should happen when a plugin is "registered" in the system. currently we are "registering" the plugins by adding them directly to the database in the patches, so we need to insert their client id/secret so they can communicate with the DB.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>